xamarin/Xamarin.Forms
# [All] Fix TextDecorations & DataTigger interaction

### Description of Change ###
Fixes an issue that makes it impossible to remove a TextDecoration that is set by a DataTrigger once it has been set.

More accurately, this change makes `BindableObject.ClearValue` behave the way it is supposed to for `TextDecorations`.

### Issues Resolved ### 
- fix #4121
- fix #8272 

### Platforms Affected ### 
 - iOS
- macOS
- Android
- UWP
- WPF

### Behavioral/Visual Changes ###
Calls to `ClearValue` on Elements with `TextDecorations` will now correctly display their pre-datatrigger appearance.

### Testing Procedure ###
Run the included UITest. The switch should cause to alternately appear and disappear.

### PR Checklist ###
